src/node_errors.cc

```diff
@@ -1044,6 +1050,304 @@ void TriggerUncaughtException(Isolate* isolate, const v8::TryCatch& try_catch) {
+        Local<FunctionTemplate>(),
+        v8::PropertyAttribute::ReadOnly);
+    tmpl->PrototypeTemplate()->SetAccessorProperty(
+        env->message_string(),
+        FunctionTemplate::New(env->isolate(), GetMessage),
+        Local<FunctionTemplate>(),
+        v8::PropertyAttribute::ReadOnly);
+    tmpl->PrototypeTemplate()->SetAccessorProperty(
+        env->stack_string(),
+        FunctionTemplate::New(env->isolate(), GetStack),
+        Local<FunctionTemplate>(),
+        v8::PropertyAttribute::ReadOnly);
+    tmpl->PrototypeTemplate()->SetAccessorProperty(
+        env->code_string(),
+        FunctionTemplate::New(env->isolate(), GetCode),
+        Local<FunctionTemplate>(),
+        v8::PropertyAttribute::ReadOnly);
+
+    env->set_domexception_constructor_template(tmpl);
+  }
+  return tmpl;
+}
+
+void DOMException::Initialize(Environment* env, v8::Local<v8::Object> target) {
+  env->SetConstructorFunction(
+      target,
+      "DOMException",
+      GetConstructorTemplate(env),
+      Environment::SetConstructorFunctionFlag::NONE);
+}
+
+void DOMException::RegisterExternalReferences(
+    ExternalReferenceRegistry* registry) {
+  registry->Register(New);
+  registry->Register(GetMessage);
+  registry->Register(GetName);
+  registry->Register(GetStack);
+  registry->Register(GetCode);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    const TransferData& transferData) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(env, obj, transferData);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    const std::string& message,
+    const std::string& name) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(env, obj, message, name);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    Local<Value> message,
+    const std::string& name) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(
+      env,
+      obj,
+      message,
+      String::NewFromUtf8(env->isolate(), name.c_str()).ToLocalChecked());
```

Comment:
```suggestion
      String::NewFromUtf8(env->isolate(), name.c_str())
          .FromMaybe(Local<String>()));
```
Why crash the process in case of an ongoing exception?